### DISCUSSION TITLE
Add "use client" to Jotai React import

### DISCUSSION THREAD

**Discussion by dallonf** [pmndrs/jotai#2241] [Ideas] [answered]
Currently you can't import the Jotai `Provider` in a server component, such as a layout. I'm currently working around this with a file that contains only:

```ts
'use client';
export { Provider as JotaiProvider } from 'jotai';
```

But it'd be nice to have this baked into the library!

This isn't about broader Jotai SSR support; I just want to add the client-side provider to the global template.

**Comment by dai-shi** [comment]
Yes, that's generally makes sense.
But, I wonder your solution isn't a big pain for the moment, no?
The "use client" directive won't be added in `jotai`, it will be in `jotai/react`.
~So, it's a broader issue about updating the entire docs.~ It's actually fine.
I like the idea very much about supporting RSC (unless people don't misunderstand it).
Let's get more feedback for now.

**Reply by himself65** [reply]
I think adding `react-server` into `exports` is better. Since I feel some old bundler might threaten it as some unknown detective (like in https://github.com/rollup/rollup/issues/4699)

**Reply by dai-shi** [reply]
I'm considering using this: https://github.com/stropho/rollup-plugin-banner2

**Reply by dai-shi** [reply, marked as answer]
#2245
